zigup.zig

```diff
@@ -421,19 +428,78 @@ fn listCompilers(allocator: *Allocator) !void {
     }
 }
 
-fn printDefaultCompiler(allocator: *Allocator) !void {
+fn cleanCompilers(allocator: *Allocator) !void {
+    // a hash map that tells what compilers to keep
+    var compilers_to_keep = std.StringHashMap(void).init(allocator);
+    defer compilers_to_keep.deinit();
+    // getting/reading the keep file
+    const install_dir_string = try getInstallDir(allocator, .{ .create = true });
+    defer allocator.free(install_dir_string);
+    const keep_file_path = try std.fs.path.join(allocator, &[_][]const u8{ install_dir_string, "keep" });
```

Comment:
The idea was that each compiler would have the keep file inside it's own install directory (not having a keep file with a list of version).

Reply:
Ah, I misread. Well this seems to be much easier to implement. Thanks